app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
Perhaps, but then this feature needs to work seamlessly for both text and binary files. 

We also have to consider what happens when the file uploaded is too big. Dropzone used in GitLab prevents users from sending anything over 10 MB, and the Web server also has to be configured appropriately to accept payloads that large.